src/aoefit.jl

```diff
@@ -327,3 +327,128 @@ function fit_single_aoe_compton(h::Histogram, ps::NamedTuple; uncertainty::Bool=
+
+Performed a combined fit over all A/E Compton band using the `f_aoe_compton` function consisting of a gaussian SSE peak and a step like background for MSE events,
+assuming `f_aoe_mu` for μ and `f_aoe_sigma` for σ.
+
+# Returns
+    * `v_ml`: The fit result from the maximum-likelihood fit.
+    * `report`: Dict of NamedTuples of the fit report which can be plotted for each compton band
+"""
+function fit_aoe_compton_combined(peakhists::Vector{<:Histogram}, peakstats::StructArray, compton_bands::Array{T}, result_corrections::NamedTuple; pars_aoe::NamedTuple{(:μ, :μ_err, :σ, :σ_err)}=NamedTuple{(:μ, :μ_err, :σ, :σ_err)}(nothing, nothing, nothing, nothing), uncertainty::Bool=false) where T<:Unitful.Energy{<:Real}
+    
+    μA = ustrip(result_corrections.μ_compton.par[1])
+    μB = ustrip(result_corrections.μ_compton.par[2])
+    σA = ustrip(result_corrections.σ_compton.par[1])
+    σB = ustrip(result_corrections.σ_compton.par[2])
+    
+    # create pseudo priors
+     pseudo_prior = NamedTupleDist(
+        μA = Normal(mvalue(μA), muncert(μA)),
+        μB = Normal(mvalue(μB), muncert(μB)),
+        σA = Normal(mvalue(σA), muncert(σA)),
+        σB = Normal(mvalue(σB), muncert(σB)),
+        #μA = Uniform(mvalue(μA) - 10*muncert(μA), mvalue(μA) + 10*muncert(μA)),
+        #μB = Uniform(mvalue(μB) - 10*muncert(μB), mvalue(μB) + 10*muncert(μB)),
+        #σA = Uniform(mvalue(σA) - 10*muncert(σA), mvalue(σA) + 10*muncert(σA)),
+        #σB = Uniform(mvalue(σB) - 10*muncert(σB), mvalue(σB) + 10*muncert(σB)),
+    )
+    
+    # transform back to frequency space
+    f_trafo = BAT.DistributionTransform(Normal, pseudo_prior)
+    
+    # start values for MLE
+    v_init = mean(pseudo_prior)
+
+    # create loglikehood function
+    f_loglike = pars -> begin
+        
+        neg_log_likelihood = 0.0
+
+        # iterate throuh all peaks
+        for (i, band) in enumerate(compton_bands)
+            # get histogram and peakstats
+            h  = peakhists[i]
+            ps = peakstats[i]
+            e = ustrip(band)
+            μ = f_aoe_mu(e, (pars.μA, pars.μB))
+            σ = f_aoe_sigma(e, (pars.σA, pars.σB))
+
+            # fit peak
+            try
+                A, B = fit_single_aoe_compton_with_fixed_μ_and_σ(h, μ, σ, ps; uncertainty=uncertainty)
+                neg_log_likelihood += A
+            catch e
+                @warn "Error fitting band $band: $e"
+                continue
+            end
+            # save results
+            # result[band] = result_band
+            # report[band] = report_band
+        end
+        return neg_log_likelihood
+    end
+    
+    # MLE
+    opt_r = optimize(f_loglike ∘ inverse(f_trafo), f_trafo(v_init))
```

Comment:
- Set `Optim.options` for number of iterations and time limit
- Check if different algorhithm should be used such as `LBFGS` maybe with `autodiff = :forward`

Reply:
I changed the line to "opt_r = optimize(f_loglike ∘ inverse(f_trafo), f_trafo(v_init), LBFGS(); autodiff=:forward)" but now it is taking forever to run the "combined_result = LegendSpecFits.fit_aoe_compton_combined(compton_band_peakhists.peakhists, compton_band_peakhists.peakstats, compton_bands, result_corrections)" line.

Reply:
Well, good to hear that you managed to get this running at all, but if it takes forever, let's leave it with `NelderMead` for now